[wesho1107] ip

wesho1107 opened the pull request:
### Birdy ChatBot

> chirp! chirp! an early bird gets the worm!  - Birdy

Birdy will help you remember things you need to do to be an early bird. It's,
- text-based
- easy to learn
- SUPER FAST to use

You just have to,
- [ ] Download it [here](https://github.com/nus-cs2103-AY2324S1/ip/pull/303)
- [ ] Double-click it
- [ ] Ask Birdy to track your tasks
- [ ] Let Birdy manage your tasks

To run Birdy, do this at the directory with your Birdy application!
```
cd <directory with your jar file>
java -jar birdy.jar
```

It is totally **free of charge** 🤩 !

Features:
1. Managing tasks: `task <taskname>`
2. Managing deadlines: `deadline <deadline> /by <duedate>`
3. Reminders for events: `event <event> /from <start_datetime> /to <end_datetime>`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

wesho1107 committed:
greet skeletal

wesho1107 committed:
added echo method and abstraction

wesho1107 committed:
added new list method, use ArrayList

wesho1107 committed:
new Task class in package, mark and unmark

wesho1107 committed:
new task classes, overriding tostring

wesho1107 committed:
new classes for Exceptions

wesho1107 committed:
automated test input/output

wesho1107 committed:
Delete diff_report.txt

wesho1107 committed:
added delete case

wesho1107 committed:
Merge branch 'master'  f github.com:wesho1107/ip

merge

wesho1107 committed:
add Read and Write functions

wesho1107 committed:
add rewrite for Delete and Mark

wesho1107 committed:
add LocalDateTime and functions

wesho1107 committed:
add DukeExceptions and Commands, OOP

wesho1107 committed:
add packages

wesho1107 committed:
add README and tests

wesho1107 committed:
Merge branch 'add-gradle-support'

add gradle support

wesho1107 committed:
change build.gradle mainClass

wesho1107 committed:
add tests for task classes

wesho1107 committed:
Add JavaDoc for methods and classes

wesho1107 committed:
Add Find function

wesho1107 committed:
Resolve merge conflict

wesho1107 committed:
Fix delete display

JasonLCY-Temp commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,46 @@
+package duke.main;
+
+import duke.command.*;
+import duke.exception.*;
```
Would listing all the imported classes explictly make the class easier to maintain in the future?

i.e.
import duke.command.AddCommand
import duke.command.Command
...

JasonLCY-Temp commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    public static void main(String[] args) {
+        new Duke("./tasks.txt").run();
```
Would using a constant variable to represent the path to the data file make it easier to comprehend the code rather than a magic String?

JasonLCY-Temp commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,72 @@
+        String headCommand = splitCommand[0];
+        String commandDetails;
+
+        if (headCommand.equals("bye")) return new ExitCommand(null);
+
+        if (headCommand.equals("list")) return new ListCommand(null);
+
+        if (splitCommand.length < 2) throw new EmptyTaskException();
```
Should the conditionals be placed on a seperate line and be wrapped with curly brackets for an easier debugging process in the future?

JasonLCY-Temp commented on src/main/java/duke/main/TaskList.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    public void add(Task task) {
+        this.tasks.add(task);
```
Is the usage of 'this' here necessary?

JasonLCY-Temp commented on src/main/java/duke/main/Ui.java:
```diff
@@ -0,0 +1,73 @@
+public class Ui {
+    private Scanner sc = new Scanner(System.in);
+    // Constants
+    private String DIVIDER = "    ____________________________________________________________\n";
+    private String LOGO = "    (• >       (• >       (• >       (• >       (• >       (• >\n"
+            +  "    /))        /))        /))        /))        /))        /))\n"
+            +  "     ``         ``         ``         ``         ``         ``\n";
+    private String HELP = "    IMPORTANT NOTES:\n"
+            + "        Todo: todo <task>\n"
+            + "        Deadline: deadline <deadline> /by <duedate>\n"
+            + "        Event: event <event> /from <start> /to <end>\n"
+            + "        Datetime format: \"dd/MM/yyyy HH:mm\"\n";
+    private String GREET = "    Hello! I'm Birdy\n"
+            + "    chirp chirp! How can I help?\n";
+    private String PARTING = "    chirp! See you around!\n";
```
Would it be better for these constants to be declared static and final as well to make it clearer that they are constants?

JasonLCY-Temp reviewed (commented):
Neatly organized code, made it easy for me to review the PR.
Just some nits that I noticed

shyanyong commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,72 @@
+        String headCommand = splitCommand[0];
+        String commandDetails;
+
+        if (headCommand.equals("bye")) return new ExitCommand(null);
```
Should conditional be put on a separate line and K&R style brackets (aka Egyptian style) be used?

shyanyong commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,72 @@
+
+        switch(headCommand) {
+            case "delete":
+                index = Integer.parseInt(commandDetails) - 1;
+                System.out.println(index);
+                return new DeleteCommand(index);
+
+            case "mark":
+                index = Integer.parseInt(commandDetails) - 1;
+                System.out.println(index);
+                return new MarkCommand(index, true);
+
+            case "unmark":
+                index = Integer.parseInt(commandDetails) - 1;
+                System.out.println(index);
+                return new MarkCommand(index, false);
+
+            case "todo":
+                task = commandDetails;
+                Todo todo = new Todo(task, false);
+                return new AddCommand(todo);
+
+            case "deadline":
+                details = commandDetails.split(" /by ");
+                if (details.length < 2) throw new NoDueDateException();
+                task = details[0];
+                String duedate = details[1];
+                Deadline deadline = new Deadline(task, duedate, false);
+                return new AddCommand(deadline);
+
+            case "event":
+                details = commandDetails.split(" /from ");
+                if(details.length < 2) throw new NoEventStartException();
+                task = details[0];
+                String[] timeDetails = details[1].split(" /to ");
+                if(timeDetails.length < 2) throw new NoEventEndException();
+                String start = timeDetails[0];
+                String end = timeDetails[1];
+                Event event = new Event(task, start, end, false);
+                return new AddCommand(event);
```
Should there be identation for `case` clauses

shyanyong commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,46 @@
+package duke.main;
```
Hey I like how your classes were split, made it very clear when looking for links between code! 👍 
I think JavaDocs should be added to some of the important methods and classes for clarity!

victorpengmx commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,72 @@
+                return new MarkCommand(index, true);
+
+            case "unmark":
+                index = Integer.parseInt(commandDetails) - 1;
```
I like how you used white space around operators. Makes your code look clean and easy to read.

victorpengmx commented on src/main/java/duke/main/Storage.java:
```diff
@@ -0,0 +1,66 @@
+                    break;
+                default:
+                    task = null;
+                    break;
```
Should there be indentation for `case` clauses?

victorpengmx commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,27 @@
+
+public class Deadline extends Task {
+    DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");
+    private LocalDateTime duedate;
```
Perhaps using camel case should be used here?

victorpengmx commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,22 @@
+import duke.main.*;
+import duke.exception.*;
+import duke.task.*;
+import java.io.IOException;
```
Should imported classes be listed explicitly?

victorpengmx reviewed (commented):
Apart from a few nits, LGTM!

wesho1107 committed:
Add JavaFX tutorial 1 to 3

wesho1107 committed:
Add DialogBox and MainWindow

wesho1107 committed:
Fix text wrap for DialogBox

wesho1107 committed:
Add assertions and egyptian brackets

wesho1107 committed:
Import statements with wildcard

Importing classes with wildcard uses larger space when some classes are not used from a library/package.

Importing classes more explicitly by specifying classes that are being used in each subclasses.

Importing classes explicitly makes the class easier to comprehend and maintain.

wesho1107 committed:
Merge pull request #1 from wesho1107/branch-A-Assertions

Merge branch-A-Assertions

wesho1107 committed:
Fix Egyptian brackets and add named constants

wesho1107 committed:
Fix switch case indentations

wesho1107 committed:
Improve code quality (named constants, dead code, complicated expressions)

Adding named constants, removing dead code and reducing complicated expressions promotes readability and maintainability of code

Constants are declared private static final, unused functions are removed and complicated conditional expressions are made into a separate function for checking

Shorten the length of code and reduce confusion when reading code

wesho1107 committed:
Merge pull request #2 from wesho1107/branch-A-CodeQuality

Improve code quality

wesho1107 committed:
Add reminders features

wesho1107 committed:
Merge branch 'master' into branch-B-Reminders

wesho1107 committed:
Merge pull request #3 from wesho1107/branch-B-Reminders

Add reminders features

wesho1107 committed:
Fix isValidIndex

wesho1107 committed:
Update JavaDoc

wesho1107 committed:
Add Ui.png

wesho1107 committed:
Add new Help feature

wesho1107 committed:
Add new Help feature

wesho1107 committed:
Add User Guide

wesho1107 committed:
Fix task test cases

wesho1107 committed:
Fix UG table format

wesho1107 committed:
Update UG spacings

wesho1107 committed:
Add _config.yml file

wesho1107 committed:
Update user guide README.md

wesho1107 committed:
Update user guide README.md

wesho1107 committed:
Fix feature list anchor links

wesho1107 committed:
Update user guide feature lists anchors

wesho1107 committed:
Update notes on user guide

wesho1107 committed:
Update dependencies for linux and windows
